cms/djangoapps/contentstore/views/course.py

```diff
@@ -1460,32 +1460,108 @@ def _get_usage_info(store, course, split_tests):
+                        continue
+
+                    unit_url = reverse_usage_url(
+                        'container_handler',
+                        course.location.course_key.make_usage_key(
+                            module_parent.location.block_type,
+                            module_parent.location.name
+                        )
+                    )
+
+                    usage_info[group_id].append({
+                        'label': u"{} / {}".format(module_parent.display_name, module.display_name),
+                        'url': unit_url,
+                    })
+        return usage_info
+
+    @staticmethod
     def update_usage_info(store, course, configuration):
         """
         Update usage information for particular Group Configuration.
 
         Returns json of particular group configuration updated with usage information.
         """
-        # Get all Experiments that use particular Group Configuration in course.
-        split_tests = store.get_items(
-            course.id,
-            category='split_test',
-            content={'user_partition_id': configuration.id}
-        )
-        configuration_json = configuration.to_json()
-        usage_information = GroupConfiguration._get_usage_info(store, course, split_tests)
-        configuration_json['usage'] = usage_information.get(configuration.id, [])
+        configuration_json = None
+        # Get all Experiments that use particular  Group Configuration in course.
+        if configuration.scheme.name == 'random':
+            split_tests = store.get_items(
+                course.id,
+                category='split_test',
+                content={'user_partition_id': configuration.id}
+            )
+            configuration_json = configuration.to_json()
+            usage_information = GroupConfiguration._get_usage_info(store, course, split_tests)
+            configuration_json['usage'] = usage_information.get(configuration.id, [])
+        elif configuration.scheme.name == 'cohort':
+            # In case if scheme is "cohort"
+            configuration_json = GroupConfiguration.update_content_group_usage_info(store, course, configuration)
         return configuration_json
 
     @staticmethod
-    def get_or_create_content_group_configuration(course):
+    def update_content_group_usage_info(store, course, configuration):
+        """
+        Update usage information for particular Content Group Configuration.
+
+        Returns json of particular content group configuration updated with usage information.
+        """
+        usage_info = GroupConfiguration.get_content_groups_usage_info(store, course)
+        content_group_configuration = configuration.to_json()
+
+        for group in content_group_configuration['groups']:
+            group['usage'] = usage_info.get(group['id'], [])
+
+        return content_group_configuration
+
+    @staticmethod
+    def get_or_create_content_group_configuration_with_usage(store, course):
```

Comment:
This might be the longest name I've ever seen! Can you go with something shorter? I don't think a content group is itself a configuration, and maybe the usage can be implied. How about ``get_or_create_content_group``, or even ``ensure_content_group``?